### PR TITLE
Update README and fix bugs

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,134 @@
...
+* [Network Setup](#network-setup)
+* [Packaging and Package Distribution](#packaging-andpackage-distribution)
+* [Contributing](#contributing)
+* [License](#license)
 
-## Contribute
+## Background
 
-Have a patch that will benefit this project? Awesome! Follow these steps to have it accepted.
+The Windows guest environment denotes the Google provided configuration and
```
Comment (zmarano) [inline README.md]: The Windows guest environment is the Google provided....
(its not just a hint)

File: README.md [hunk 2]
```diff
@@ -1,27 +1,134 @@
...
+tooling inside of a [Google Compute Engine](https://cloud.google.com/compute/)
+(GCE) virtual machine. The
+[metadata server](https://cloud.google.com/compute/docs/metadata) is a
+communication channel for transferring information from a client into the guest.
+The Windows guest environment includes a set of scripts and binaries that read 
+the content of the metadata server to make a virtual machine run properly on our 
+platform.
```
Comment (zmarano) [inline README.md]: Google Compute Engine (or the platform).

File: README.md [hunk 3]
```diff
@@ -1,27 +1,134 @@
...
+## Agent
 
-## License
+#### Account Setup
+
+The agent handles [creating user accounts and setting/resetting passwords](https://cloud.google.com/compute/docs/instances/windows/creating-passwords-for-windows-instances).
+
+#### IP Forwarding
+
+The agent uses IP forwarding metadata to setup or remove IP routes in the guest.
```
Comment (zmarano) [inline README.md]: IP routes. (remove in the guest)

File: README.md [hunk 4]
```diff
@@ -1,27 +1,134 @@
...
+
+*   Only IPv4 IP addresses are currently supported.
+
+## Instance Setup
+
+`instance_setup.ps1` is configured by GCE sysprep to run on VM first boot. The script
```
Comment (zmarano) [inline README.md]: The script performs the following tasks:

File: README.md [hunk 5]
```diff
@@ -1,27 +1,134 @@
...
+The agent uses IP forwarding metadata to setup or remove IP routes in the guest.
+
+*   Only IPv4 IP addresses are currently supported.
+
+## Instance Setup
+
+`instance_setup.ps1` is configured by GCE sysprep to run on VM first boot. The script 
+configures the Windows guest environment by performing the following tasks.
+
+*   Setting Hostname
```
Comment (zmarano) [inline README.md]: + Set the hostname to the the instance name.
+ Runs user provided 'specialize' startup script.
+ Activates Windows using a KMS server.
+ Sets up RDP and WinRM to allow remote login.

File: README.md [hunk 6]
```diff
@@ -1,27 +1,134 @@
...
+*   Running user provided 'specialize' startup script
+*   Activating Windows
+*   Seting up RDP and WinRM
+
+## Metadata Scripts
+
+Metadata scripts implement support for running user provided
+[startup scripts](https://cloud.google.com/compute/docs/startupscript) and
+[shutdown scripts](https://cloud.google.com/compute/docs/shutdownscript).
+
```
Comment (zmarano) [inline README.md]: Add in types of scripts (bat and ps1) and list the metadata key values.

File: README.md [hunk 7]
```diff
@@ -1,27 +1,134 @@
...
+## Packaging and Package Distribution
+
+The guest code is packaged in [GooGet](https://github.com/google/googet)
+packages and published to Google Cloud repositories, if you are creating 
+a custom image, you can also use these repositories in your image.
```
Comment (zmarano) [inline README.md]: Remove "if you are creating a customer image..." for now as that is not yet a supported use case for Windows on GCE.

File: README.md (hunk 6 above)
Comment (adjackura) [inline README.md]: Are the links to the actual documentation not enough? If I list everything then the detailed documentation will exist in two places.

Comment (zmarano) [inline README.md]: Yeah that's fine. It is nice to actual document what the code is doing in the code docs though. Likewise for any metadata key we're using here.

File: README.md [hunk 8]
```diff
@@ -1,27 +1,133 @@
...
+## Packaging and Package Distribution
+
+The guest code is packaged in [GooGet](https://github.com/google/googet)
+packages and published to Google Cloud repositories.
 
-All files in this repository are under the [Apache License, Version 2.0](LICENSE) unless noted otherwise.
+We build and install the following packages for the Windows guest environment:
+  
+*   `google-compute-engine-windows` 
+    *   Windows agent exe
```
Comment (illfelder) [inline README.md]: Can you drop the internal bullets and make them sentences?

*   `google-compute-engine-windows` - Windows agent executable.

**Pull request merged**